tests/fiaas_deploy_daemon/deployer/kubernetes/test_service_account_deploy.py

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env python
+# -*- coding: utf-8
+
+# Copyright 2017-2019 The FIAAS Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import pytest
+from mock import create_autospec
+from requests import Response
+
+from k8s.models.service_account import ServiceAccount
+
+from fiaas_deploy_daemon.config import Configuration
+from fiaas_deploy_daemon.deployer.kubernetes.service_account import ServiceAccountDeployer
+from fiaas_deploy_daemon.specs.models import LabelAndAnnotationSpec
+
+from utils import TypeMatcher
+
+SERVICES_ACCOUNT_URI = '/api/v1/namespaces/default/serviceaccounts/'
+LABELS = {"service": "pass through"}
+
+class TestServiceDeployer(object):
+#    @pytest.fixture(params=("ClusterIP", "NodePort", "LoadBalancer"))
+#    def service_type(self, request):
+#        return request.param
+
+    @pytest.fixture
+    def deployer(self, owner_references):
+        config = create_autospec(Configuration([]), spec_set=True)
+        return ServiceAccountDeployer(config, owner_references)
+
+    @pytest.mark.usefixtures("get")
+    def test_deploy_new_service_account(self, deployer, post, app_spec, owner_references):
```

Comment:
With the new logic about dealing with an existing account with the app's name, and copying imagePullSecrets from default, I think a couple more tests to handle those extra cases would be good to see:
 * if the account exists and isn't owned by fiaas, verify no calls are made
 * when the default account has imagePullSecrets, they are present in the new account that's created

Reply:
Would you be kind enough to write the later? (looking for inspiration :p)